src/main/java/fi/pohina/vinkkilista/data_access/PostgresBookmarkDao.java

```diff
@@ -148,22 +102,13 @@ private String createTag(String tagName) {
      * @param tagName
      * @param bookmarkId
      */
-    private void addBookmarkTag(String tagName, String bookmarkId) {
-        Tag tag = getTag(tagName);
+    private void addBookmarkTag(String bookmarkId, String tagId) {
         String query = "INSERT INTO bookmark_tags (\"bookmarkId\", \"tagId\") values (?, ?)";
         try {
-            if (tag != null) {
```

Comment:
This is needed. If the tag exists in the database, don't create a new tag. If the tag doesn't exists in the database, it needs to be created for the relations to work.

Reply:
I believe that the service class checks via TagDao whether any suggested tags exist, if not they are created there. I think by adding tag creation to BookmarkDao, the two Dao classes will share responsibilities that may not be necessary? Or do you suggest BookmarkDao to know TagDao so that the latter is called when a new tag needs to be created?